pkgs/objective_c/lib/src/ns_input_stream.dart

```diff
@@ -15,23 +27,59 @@ extension NSInputStreamStreamExtension on Stream<List<int>> {
   /// > than the one that calls [toNSInputStream]. Otherwise,
   /// > [NSInputStream.read] will deadlock waiting for data to be added from the
   /// > [Stream].
+  ///
+  /// > [!IMPORTANT]
+  /// > [NSInputStream.read] must be called from a different thread or [Isolate]
+  /// > than the one that calls `toNSInputStream`. Otherwise,
+  /// > [NSInputStream.read] will deadlock waiting for data to be added from the
+  /// > [Stream].
+  ///
+  /// > [!IMPORTANT]
+  /// > `toNSInputStream` creates a reference cycle between Dart and
+  /// > Objective-C. Unless this cycle is broken, the [Isolate] calling
+  /// > `toNSInputStream` will never exit. The cycle can be broken by calling
+  /// > [NSInputStream.close].
+  // TODO(brianquinlan): Add a note saying that the Dart/Objective-C reference
```

Comment:
Are you planning to resolve this TODO in this PR?

Reply:
Oh, my bad. I updated the docs. WDYT?